frontend/src/metabase/containers/SaveQuestionModal/SaveQuestionForm.tsx

```diff
@@ -0,0 +1,70 @@
+import { TransitionGroup } from "react-transition-group";
+import { t } from "ttag";
+
+import FormCollectionPicker from "metabase/collections/containers/FormCollectionPicker";
+import { useSaveQuestionContext } from "metabase/containers/SaveQuestionModal/context";
+import type { SaveQuestionFormProps } from "metabase/containers/SaveQuestionModal/types";
+import Button from "metabase/core/components/Button";
+import FormErrorMessage from "metabase/core/components/FormErrorMessage";
+import FormFooter from "metabase/core/components/FormFooter";
+import FormInput from "metabase/core/components/FormInput";
+import FormRadio from "metabase/core/components/FormRadio";
+import FormSubmitButton from "metabase/core/components/FormSubmitButton";
+import FormTextArea from "metabase/core/components/FormTextArea";
+import CS from "metabase/css/core/index.css";
+import { Form } from "metabase/forms";
+import { DEFAULT_MODAL_Z_INDEX } from "metabase/ui";
+
+export const SaveQuestionForm = ({ onCancel }: SaveQuestionFormProps) => {
+  const { nameInputPlaceholder, originalQuestion, showSaveType, values } =
```

Comment:
in my opinion, I would move `nameInputPlaceholder` and `cardType` to this component as we can derive it from `getPlaceholder(question.type())` and `question.type()` respectively - if we can derive a value from a context value, then we would not need that in the context.

based on this line, I think `question`, `originalQuestion` and `values` would make sense to be in the context as it can be reused effectively across components